app/rpc.js

```diff
@@ -419,11 +419,18 @@ export default class RPC {
   // Get all Addresses, including T and Z addresses
   async fetchAllAddresses() {
     const zaddrsPromise = RPC.doRPC('z_listaddresses', [], this.rpcConfig);
-    const taddrsPromise = RPC.doRPC('getaddressesbyaccount', [''], this.rpcConfig);
-
+    const taddrs1Promise = RPC.doRPC('getaddressesbyaccount', [''], this.rpcConfig);
+    const taddrs2Promise = RPC.doRPC('listaddressgroupings', [], this.rpcConfig);
```

Comment:
Can you please replace this code with
```
    const zaddrsPromise = RPC.doRPC('z_listaddresses', [], this.rpcConfig);
    const taddrs1Promise = RPC.doRPC('getaddressesbyaccount', [''], this.rpcConfig);
    const taddrs2Promise = RPC.doRPC('listaddressgroupings', [], this.rpcConfig);

    const allZ = (await zaddrsPromise).result;

    const t1 = (await taddrs1Promise).result;
    const t2 = (await taddrs2Promise).result.map(a => a[0]).map(a => a[0]);
    const allT = [...new Set(t1.concat(t2))];

    this.fnSetAllAddresses(allZ.concat(allT));
```

Reply:
Using 2 for loops and a linear search (`indexof`) is likely to be very slow if the user has a lot of addresses. This code is called for every refresh, so probably best to optimize it a bit.

Reply:
Will do! Thanks for the fix, js not my strong suite but just whipped something together that worked. Replaced with that code, tested and indeed works as well.